Apps/Alarm_controller/alarm_controller.c

```diff
@@ -42,92 +43,76 @@ void AlarmController_Init(void)
  ******************************************************************************/
 void AlarmController_Loop(void)
 {
-    static short previous_id       = -1;
     static uint8_t blink           = 0;
     static uint8_t blink_nb        = BLINK_NUMBER * 2;
     static uint32_t last_blink     = 0;
-    static uint32_t detection_date = 0;
+    static uint32_t last_detection = 0;
 
     // ********** hot plug management ************
     // Check if we have done the first init or if service Id have changed
-    if (previous_id != RoutingTB_IDFromService(app))
+    if (Luos_IsNodeDetected())
     {
-        if (RoutingTB_IDFromService(app) == 0)
+        if (end_detection)
```

Comment:
I think all the configurations should go in the message_handler, on the `if (msg->header.cmd == END_DETECTION)`